Marlin/Marlin_main.cpp

```diff
@@ -2842,98 +2887,43 @@ inline void gcode_G28() {
 
     #elif defined(MIN_Z_HEIGHT_FOR_HOMING) && MIN_Z_HEIGHT_FOR_HOMING > 0
 
-      #if HAS_BED_PROBE
-        do_probe_raise(MIN_Z_HEIGHT_FOR_HOMING);
-        destination[Z_AXIS] = current_position[Z_AXIS];
-      #else
-        // Raise Z before homing any other axes and z is not already high enough (never lower z)
-        if (current_position[Z_AXIS] <= MIN_Z_HEIGHT_FOR_HOMING) {
-          destination[Z_AXIS] = MIN_Z_HEIGHT_FOR_HOMING;
-          feedrate = planner.max_feedrate[Z_AXIS] * 60;  // feedrate (mm/m) = max_feedrate (mm/s)
-          #if ENABLED(DEBUG_LEVELING_FEATURE)
-            if (DEBUGGING(LEVELING)) {
-              SERIAL_ECHOPAIR("Raise Z (before homing) to ", (MIN_Z_HEIGHT_FOR_HOMING));
-              SERIAL_EOL;
-              DEBUG_POS("> (home_all_axis || homeZ)", current_position);
-              DEBUG_POS("> (home_all_axis || homeZ)", destination);
-            }
-          #endif
-          line_to_destination();
-          stepper.synchronize();
-
-          /**
-           * Update the current Z position even if it currently not real from
-           * Z-home otherwise each call to line_to_destination() will want to
-           * move Z-axis by MIN_Z_HEIGHT_FOR_HOMING.
-           */
-          current_position[Z_AXIS] = destination[Z_AXIS];
+      // Raise Z before homing any other axes and z is not already high enough (never lower z)
+      destination[Z_AXIS] += MIN_Z_HEIGHT_FOR_HOMING;
+      #if ENABLED(DEBUG_LEVELING_FEATURE)
+        if (DEBUGGING(LEVELING)) {
+          SERIAL_ECHOPAIR("Raise Z (before homing) to ", destination[Z_AXIS]);
+          SERIAL_EOL;
         }
       #endif
-    #endif
-
-    #if ENABLED(QUICK_HOME)
-
-      if (home_all_axis || (homeX && homeY)) {  // First diagonal move
-
-        current_position[X_AXIS] = current_position[Y_AXIS] = 0;
-
-        #if ENABLED(DUAL_X_CARRIAGE)
-          int x_axis_home_dir = x_home_dir(active_extruder);
-          extruder_duplication_enabled = false;
-        #else
-          int x_axis_home_dir = home_dir(X_AXIS);
-        #endif
-
-        SYNC_PLAN_POSITION_KINEMATIC();
+      do_blocking_move_to_z(destination[Z_AXIS]);
```

Comment:
If `MIN_Z_HEIGHT_FOR_HOMING` is defined you get.
```
Marlin_main.cpp:2898: error: 'do_blocking_move_to_z' was not declared in this scope

       do_blocking_move_to_z(destination[Z_AXIS]);
```
Because `do_blocking_move_to_z()` is only defined if  `HAS_BED_PROBE`